From b41058b970a4eba2ba962c60619e4bc4f932527c Mon Sep 17 00:00:00 2001
From: Srinivas Neeli <srinivas.neeli@xilinx.com>
Date: Fri, 31 Jul 2020 14:03:40 +0530
Subject: [PATCH 1497/1851] gpio: gpio-xilinx: Check return value of
 irq_get_handler_data

commit ff86ae0742da315f3f0f575a1462837f8f9cde8c from
https://github.com/Xilinx/linux-xlnx.git

Check return value of irq_get_handler_data for error.

Addresses-Coverity: "returned_null"
Signed-off-by: Srinivas Neeli <srinivas.neeli@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpio/gpio-xilinx.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpio/gpio-xilinx.c b/drivers/gpio/gpio-xilinx.c
index 3467cc86fd7f..d94af260ea1d 100644
--- a/drivers/gpio/gpio-xilinx.c
+++ b/drivers/gpio/gpio-xilinx.c
@@ -411,6 +411,9 @@ static void xgpio_irqhandler(struct irq_desc *desc)
 	int offset;
 	unsigned long val;
 
+	if (!chip)
+		return;
+
 	chained_irq_enter(irqchip, desc);
 
 	val = xgpio_readreg(mm_gc->regs + chip->offset);
-- 
2.31.1

